src/editor/initialization/keybindings.js

```diff
@@ -1,18 +1,36 @@
 import Mousetrap from "mousetrap";
 import tasks from "../user-tasks/task-definitions";
-// for now, let's just keep the keybindings here.
 
 Mousetrap.prototype.stopCallback = () => false;
 
+let warnUser = false;
+
+const preventBacknav = e => {
+  if (e.target === document.body) {
+    warnUser = true;
+  } else {
+    warnUser = false;
+  }
+};
+
 export function initializeDefaultKeybindings() {
   Object.keys(tasks).forEach(t => {
     const task = tasks[t];
     if (task.hasKeybinding()) {
       Mousetrap.bind(task.keybindings, task.keybindingCallback);
     }
   });
+  Mousetrap.bind(["delete", "backspace"], preventBacknav);
 }
 
+window.onbeforeunload = () => {
+  if (warnUser) {
+    warnUser = false;
+    return "Are you sure you want to leave?";
+  }
+  return undefined;
+};
+
```

Comment:
this approach allows us to only pop up the browser `beforeunload` modal if a user hits backspace. I could have this modal trigger _any_ time we try to unload from the client (that includes browser refresh) - thoughts here woudl be appreciated. Always triggering this on unload would simplify things quite a bit I think.

Reply:
the simplification would be just not having the mousetrap.bind and the `preventBacknav`, or is there something else?

Reply:
i think that simplifcation seems like the right thing actually -- let's just always warn before unload. let's make it hard for users to accidentally lose state.